### PR TITLE
Avoid using Symbol::intern

### LINKED ISSUE #4266: cargo fmt stopped working since 1.44.0
I'm trying to format code for the [druid](https://github.com/xi-editor/druid) project.
Everything was working fine up until Rust 1.43.1 / rustfmt 1.4.12-stable (a828ffea 2020-03-11).

It no longer works properly with:
Rust 1.44.0 / rustfmt 1.4.14-stable (e4173566 2020-04-21)
Rust 1.44.1 / rustfmt 1.4.16-stable (939e164c 2020-06-11)

It does not seem to be platform dependant, the issue exists at minimum both on Windows 7 and macOS Catalina, and not just my machines.

The druid workspace consists of several packages including [`druid-shell`](https://github.com/xi-editor/druid/tree/master/druid-shell) which is the package that is problematic.

If I run `cargo fmt` inside the `druid-shell` package root then it works.

The problem is with using `cargo fmt` or `cargo fmt --all` in the workspace directory.
That used to also work, but now works only partially on `druid-shell`.

Some of the modules get formatted while others don't.
It seems to have something to do with the [`cfg-if`](https://crates.io/crates/cfg-if) macro.

A file of interest is [`druid-shell/src/platform/mod.rs`](https://github.com/xi-editor/druid/blob/master/druid-shell/src/platform/mod.rs), which has the following code:

```rust
cfg_if::cfg_if! {
    if #[cfg(target_os = "windows")] {
        mod windows;
        pub use windows::*;
    } else if #[cfg(target_os = "macos")] {
        mod mac;
        pub use mac::*;
    } else if #[cfg(all(feature = "x11", target_os = "linux"))] {
        mod x11;
        pub use x11::*;
    } else if #[cfg(target_os = "linux")] {
        mod gtk;
        pub use self::gtk::*;
    } else if #[cfg(target_arch = "wasm32")] {
        mod web;
        pub use web::*;
    }
}
```

If I delete that macro and replace it with a simple:
```rust
mod windows;
pub use windows::*;
```
Then the `windows` module gets formatted when I run `cargo fmt` in the workspace directory. However if the `cfg-if` macro is there, then none of these platform modules get formatted. Again, this is a recent regression and previously all those platform modules got formatted.

Any insight would be appreciated.

### PR DESCRIPTION
**Pull request by topecongiro**
Close #4266.

### CODE REVIEW

File: src/formatting/modules/visitor.rs [hunk 1]
```diff
@@ -49,9 +49,9 @@ impl<'a, 'ast: 'a> CfgIfVisitor<'a> {
         // extern crate cfg_if;
         // cfg_if! {..}
         // ```
-        match mac.path.segments.first() {
-            Some(first_segment) => {
-                if first_segment.ident.name != Symbol::intern("cfg_if") {
+        match mac.path.segments.last() {
```
Comment (calebcartwright) [inline src/formatting/modules/visitor.rs:52]: I assume the first to last change will still allow support for `cfg_if::cfg_if!` and  `cfg_if!` correct?

Comment (topecongiro) [reply src/formatting/modules/visitor.rs:52]: Yes, I added to support self-importing like `pub use cfg_if::cfg_if;` in the root file and using cfg_if! as `crate::cfg_if!`.

Comment (calebcartwright) [reply src/formatting/modules/visitor.rs:52]: Even better! Changes LGTM, CI failures seem to just be due to the configuration snippet tests needing lazy_static

Comment (topecongiro) [reply src/formatting/modules/visitor.rs:52]: Thanks for the catch! Fixed in https://github.com/rust-lang/rustfmt/pull/4268/commits/5c7cb944697426a8e6139c438e457c91e14cec79.

**Pull request merged**